docs/src/use/migrate-to-10.0.0.md

```diff
@@ -148,3 +149,28 @@ For example, this configuration is now invalid due to the extra element `"foo"`:
     - optionally, an object option: `{ "generators": "always" | "as-needed" | "never" }`.
 
 **Related issue(s):** [#20134](https://github.com/eslint/eslint/issues/20134)
+
+## <a name="stricter-rule-tester"></a> Valid test cases must not have `errors` or `output` properties
+
+In ESLint v10, the RuleTester has become more strict about test case structure. Valid test cases (those that should not produce any linting errors) are no longer allowed to have `errors` or `output` properties.
+​What changed:​​
+
+- Previously, valid test cases could include `errors` or `output` properties, which were ignored
+- Now, including these properties in valid test cases will cause the test to fail
+
+Example of invalid usage:​
+
+```js
+// This will now throw an error in ESLint v10
+const validTestCases = [
```

Comment:
I think this would be helpful to give the context with `ruleTester.run("rule-id", rule", { valid: [...] })`.